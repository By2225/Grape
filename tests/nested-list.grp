fun Int main() {
        String hi = "hi";
        prints(hi[0]);

        List<List<Int> > a = [
                [1,2,3,4,5],
                [1,2,3,4,5],
                [1,2,300,4,5],
                [1,2,3,4,5]];

        print(a[2][2]);

        List<List<Node<Int> > > b = [
                ['1','2','3','4','5'],
                ['1','2','3','4','5'],
                ['1','2','313','4','5'],
                ['1','2','3','4','5']];

        print(b[2][2].val);

        return 0;
}
