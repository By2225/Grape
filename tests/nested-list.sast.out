
fun Int main() {
String (String : hi = (String : hi));
(* : print((String : (String : hi)[(Int : 0)])));
List<List<Int>> (List<List<Int>> : a = (List<List<Int>> : [(List<Int> : [(Int : 1),(Int : 2),(Int : 3),(Int : 4),(Int : 5)]),(List<Int> : [(Int : 1),(Int : 2),(Int : 3),(Int : 4),(Int : 5)]),(List<Int> : [(Int : 1),(Int : 2),(Int : 300),(Int : 4),(Int : 5)]),(List<Int> : [(Int : 1),(Int : 2),(Int : 3),(Int : 4),(Int : 5)])]));
(* : print((Int : (List<Int> : (List<List<Int>> : a)[(Int : 2)])[(Int : 2)])));
List<List<Node<Int>>> (List<List<Node<Int>>> : b = (List<List<Node<Int>>> : [(List<Node<Int>> : [(Node<Int> : '(Int : 1)'),(Node<Int> : '(Int : 2)'),(Node<Int> : '(Int : 3)'),(Node<Int> : '(Int : 4)'),(Node<Int> : '(Int : 5)')]),(List<Node<Int>> : [(Node<Int> : '(Int : 1)'),(Node<Int> : '(Int : 2)'),(Node<Int> : '(Int : 3)'),(Node<Int> : '(Int : 4)'),(Node<Int> : '(Int : 5)')]),(List<Node<Int>> : [(Node<Int> : '(Int : 1)'),(Node<Int> : '(Int : 2)'),(Node<Int> : '(Int : 313)'),(Node<Int> : '(Int : 4)'),(Node<Int> : '(Int : 5)')]),(List<Node<Int>> : [(Node<Int> : '(Int : 1)'),(Node<Int> : '(Int : 2)'),(Node<Int> : '(Int : 3)'),(Node<Int> : '(Int : 4)'),(Node<Int> : '(Int : 5)')])]));
(* : print((Int : (Node<Int> : (List<Node<Int>> : (List<List<Node<Int>>> : b)[(Int : 2)])[(Int : 2)]).val)));
return (Int : 0);
}
